### PR TITLE
Add support for submitting standalone entry points to Azure Quantum

### PR DESCRIPTION
**Pull request by samarsha**

### CODE REVIEW

File: src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs [hunk 1]
```diff
@@ -481,6 +481,9 @@ let ``Supports default custom simulator`` () =
     given ["--simulator"; typeof<QuantumSimulator>.FullName; "--use-h"; "false"] |> fails
 
 
+// TODO: Add tests for the "submit" command.
```
Comment (bettinaheim) [inline src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs:484]: :) 
What's the plan for tests? Are there any test planned on the runtime repo, or will this be tested as part of our service tests?

File: src/Simulation/EntryPointDriver/Azure.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                return 1;
+            }
+
+            var job = await machine.SubmitAsync(entryPoint.Info, entryPoint.CreateArgument(parseResult));
+            switch (settings.Output)
+            {
+                case OutputFormat.FriendlyUri:
+                    Console.WriteLine("Job submitted. To track your job status and see the results use:");
+                    Console.WriteLine();
+                    // TODO: Show the friendly URI.
```
Comment (bettinaheim) [inline src/Simulation/EntryPointDriver/Azure.cs:40]: What's this is a todo rather than done?

File: src/Simulation/EntryPointDriver/Azure.cs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        /// <summary>
+        /// Creates a quantum machine based on the Azure Quantum submission settings.
+        /// </summary>
+        /// <param name="settings">The Azure Quantum submission settings.</param>
+        /// <returns>A quantum machine.</returns>
+        private static IQuantumMachine? CreateMachine(AzureSettings settings)
+        {
+            if (!(settings.Target is null) && settings.Target.StartsWith("ionq."))
+            {
+                // TODO: Give the number of shots to the quantum machine.
```
Comment (bettinaheim) [inline src/Simulation/EntryPointDriver/Azure.cs:61]: same question here

File: src/Simulation/EntryPointDriver/Simulation.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        /// <param name="entryPoint">The entry point.</param>
+        /// <param name="parseResult">The command-line parsing result.</param>
+        /// <param name="createSimulator">A function that creates an instance of the simulator to use.</param>
+        private static async Task RunSimulator(
+            IEntryPoint<TIn, TOut> entryPoint, ParseResult parseResult, Func<IOperationFactory> createSimulator)
+        {
+            var simulator = createSimulator();
+            try
+            {
+                var value = await simulator.Run<TCallable, TIn, TOut>(entryPoint.CreateArgument(parseResult));
```
Comment (bettinaheim) [inline src/Simulation/EntryPointDriver/Simulation.cs:67]: @anpaz-msft Is invoking .Run the recommended way for executing? I vaguely remember some other method having been added related to getting a better stack trace handling?

File: src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs (hunk 1 above)
Comment (samarsha) [reply src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs:484]: My plan is to test all of the command-line option handling in the runtime repo using the `NothingMachine` as the target. For other quantum machines like IonQ, I think those will have to go in another repository where those machines are available by a direct reference.

File: src/Simulation/EntryPointDriver/Azure.cs (hunk 2 above)
Comment (samarsha) [reply src/Simulation/EntryPointDriver/Azure.cs:40]: There is not a way to get the friendly URI from the job at the moment. I think @cesarzc might have something planned for that?

File: src/Simulation/EntryPointDriver/Azure.cs (hunk 3 above)
Comment (samarsha) [reply src/Simulation/EntryPointDriver/Azure.cs:61]: The `IQuantumMachine` interface does not support specifying the number of shots when a job is submitted. @cesarzc was planning to update it.

File: src/Simulation/EntryPointDriver/Simulation.cs (hunk 4 above)
Comment (samarsha) [reply src/Simulation/EntryPointDriver/Simulation.cs:67]: When C# generation generates a `Run` method for operations, it calls `IOperationFactory.Run`, so I was just following that.

File: src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs (hunk 1 above)
Comment (samarsha) [reply src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs:484]: Tests are in #204.

Comment (bettinaheim) [reply src/Simulation/CsharpGeneration.Tests/EntryPointTests.fs:484]: sounds good.

File: src/Simulation/EntryPointDriver/Azure.cs (hunk 2 above)
Comment (bettinaheim) [reply src/Simulation/EntryPointDriver/Azure.cs]: Ah, I see. Just curious.

File: src/Simulation/EntryPointDriver/Simulation.cs (hunk 4 above)
Comment (bettinaheim) [reply src/Simulation/EntryPointDriver/Simulation.cs:67]: Ok, let's keep with that then - it's a quick thing to change in any case.

**Pull request merged**